controllers/workloads.go

```diff
@@ -260,6 +260,18 @@ func (n *Workloads) Reconcile(ctx context.Context, clt client.Client, scheme *ru
 
 	log := ctrllog.FromContext(ctx)
 
+	namespace, err := getNamespace()
+	if err != nil {
+		log.Error(err, "failed to determine which namespace mondoo-operator is running in")
+		return ctrl.Result{}, err
+	}
+
+	if n.Mondoo.Spec.Workloads.ServiceAccount == "" && n.Mondoo.Namespace == namespace {
+		n.Mondoo.Spec.Workloads.ServiceAccount = defaultServiceAccount
+	} else {
+		log.Info("MondooAuditConfig in different namespace than mondoo-operator requires setting Workloads.ServiceAccount")
+	}
+
```

Comment:
I believe this should simply be an error/early exit. There is no point in trying to proceed as we don't know which ServiceAccount to bind to the workload Deployment when the MondooAuditConfig is in a different namespace than mondoo-operator.

```suggestion
	if n.Mondoo.Spec.Workloads.ServiceAccount == "" && n.Mondoo.Namespace == namespace {
		n.Mondoo.Spec.Workloads.ServiceAccount = defaultServiceAccount
	}
	if n.Mondoo.Spec.Workloads.ServiceAccount == "" {
	       err := fmt.Errorf("MondooAuditConfig.spec.workloads.serviceAccount cannot be empty when running in a different namespace than mondoo-operator")
		log.Error(err, "ServiceAccount cannot be empty")
		return ctlr.Result{}, err
	}
```